components/tableIndexer/table_constructor.py

```diff
@@ -0,0 +1,154 @@
+from transformers import TableTransformerForObjectDetection
+from utils.logger import setup_logging
+import logging
+import argparse
+import os
+import json as js
+import sys
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Construct tables by defining rows for detected tables."
+    )
+    parser.add_argument(
+        "--log-level",
+        dest="log_level",
+        default="INFO",
+        help="Set the logging level (DEBUG, INFO, WARNING, ERROR, CRITICAL)",
+    )
+    parser.add_argument(
+        "--img-path",
+        dest="img_path",
+        required=True,
+        help="Specify the path of the cropped table",
+    )
+    parser.add_argument(
+        "--model-name",
+        dest="model_name",
+        default="microsoft/table-structure-recognition-v1.1-all",
+        help="Specify the path of the cropped table",
+    )
+    parser.add_argument(
+        "--device",
+        dest="device",
+        default="cpu",
+        help="Specify the device type to load the model (cpu, cuda, mkldnn, opengl, opencl, ideep, hip, msnpu",
+    )
+    parser.add_argument(
+        "--output-directory",
+        dest="output_directory",
+        default="./outputs/constructed_tables",
+        help="Specify the output directory for images",
+    )
+    parser.add_argument(
+        "--json", dest="json", default=False, help="Save output as a json file"
+    )
+    parser.add_argument(
+        "--json-path",
+        dest="json_path",
+        default="./outputs/constructed_tables_json",
+        help="output directory for the json file",
+    )
+    return parser.parse_args()
+
+
+def table_constructor(
+    img_path,
+    model_name,
+    device,
+    output_directory="./outputs/constructed_tables",
+    json=False,
+    json_path="./outputs/constructed_tables_json",
+):
+    try:
+        logging.info("✨ Sentinel Table Constructor has started...")
+
+        try:
+            model = TableTransformerForObjectDetection.from_pretrained(model_name)
+        except Exception as e:
+            logging.error(f"There was an issue when loading the model: {e}")
+            sys.exit(1)
+
+        logging.debug("✔️  Model Loaded")
+        logging.debug(f"🟡 Model name: {model_name}")
+        logging.debug(f"🟡 Device type: {device}")
+
+        image = Image.open(img_path).convert("RGB")
+
+        logging.debug("✔️  Image Loaded")
+
+        structure_transform = transforms.Compose(
+            [
+                MaxResize(1000),
+                transforms.ToTensor(),
+                transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]),
+            ]
+        )
+
+        pixel_values = structure_transform(image).unsqueeze(0)
+        pixel_values = pixel_values.to(device)
+
+        with torch.no_grad():
+            outputs = model(pixel_values)
+
+        structure_id2label = model.config.id2label
+        structure_id2label[len(structure_id2label)] = "no object"
```

Comment:
**Suggestion: Use an enum here**


```python
class LabelType(enum):
       NO_OBJECT ...
```

or something akin to that

makes it a little more explicit what values it can be